juriscraper/pacer/activity_report.py

```diff
@@ -0,0 +1,254 @@
+        #             <td valign="top">ARC<br>
+        #             <i>Type: </i>crt</td>
+        #             <td valign="top"><i>Office:</i> Tucson Division
+        #             <br><i>Presider: </i>Eric J Markovich
+        #             </td>
+        #         </tr>
+        #         <tr>
+        #             <td valign="top"><a href="https://ecf.azd.uscourts.gov/cgi-bin/DktRpt.pl?1224861">4:20-po-20410-EJM <br>USA v. Florentino-Cano<br></a><b><font size="-1"> CASE CLOSED on 01/10/2020</font></b></td>
+        #             <td valign="top"><i>Entered: </i>01/13/2020 <br>15:45:25<br><i>Filed: </i>01/13/2020</td>
+        #             <td valign="top"><i>Category:</i> order-cr<br><i>Event:</i> OSL Order - Petty Cases Only<br><i>Document: </i><a href="https://ecf.azd.uscourts.gov/doc1/025121589269" onclick="goDLS(&#39;/doc1/025121589269&#39;,&#39;1224862&#39;,&#39;5&#39;,&#39;&#39;,&#39;&#39;,&#39;1&#39;,&#39;&#39;,&#39;&#39;);return(false);">1</a></td>
+        #             <td valign="top">BAR<br>
+        #             <i>Type: </i>crt</td>
+        #             <td valign="top"><i>Office:</i> Tucson Division
+        #             <br><i>Presider: </i>Eric J Markovich
+        #             </td>
+        #         </tr>
+        #     </tbody>
+        # </table>
+
+        data = []
+
+        table = self.tree.xpath('//table')[0]
+
+        rows = table.xpath('.//tr')
+
+        for b, row in enumerate(rows):
+
+            cols = row.xpath('.//node()')
+
+            def isFlags(cols):
+                #Check for case flags. It's not always present.
+                try:
+                    cols.index('Case Flags:')
+                except ValueError:
+                    flag = "none"
+                    return flag
+                else:
+                    flag = cols[cols.index('Case Flags:')+3]
+                    return flag
+
+            flags = isFlags(cols)
+
+            if b > 0: # skip first
+                data.append({
+                    u"case":cols[2],
+                    u"case_number":cols[4],
+                    u"docket_report_url":cols[1].xpath('@href')[0],
+                    u"entered": cols[cols.index('Entered: ')+1],
+                    u"time": cols[cols.index('Entered: ')+3],
+                    u"filed": cols[cols.index('Filed: ')+1],
+                    u"closed": cols[8],
+                    u"event": cols[cols.index('Event:')+1],
+                    u"case_flags": flags,
+                    u"category": cols[cols.index('Category:')+1],
+                    u"presider": cols[cols.index('Presider: ')+1],
+                    u"office": cols[cols.index('Office:')+1],
+                    u"docketed_by":cols[cols.index('Document: ')+5],
+                    u"doc1":cols[cols.index('Document: ')+1].xpath('@href')[0],
+                    u"doc_count":cols[cols.index('Document: ')+2]
```

Comment:
It's hard to understand here what you're doing with the math and the indexing and whatnot. Could you pull some of these `cols.index(xxx)` bits into usefully named variables, maybe?

Formatting wise, these should have spaces around the `+` signs. Also, no whitespace before the tailing `}`.